modin/data_management/functions/groupby_function.py

```diff
@@ -181,6 +181,11 @@ def caller(
             **kwargs,
         )
 
+        if kwargs.get("method", None) == "size":
```

Comment:
I do not like special casing this in the function object. This `GroupByFunction` object is already too large.

Can we move handling this to a higher layer?

Reply:
If we do it at the front, then at the backend, we may miss the `by` columns. This is a problem because groupby at the backend still doesn't have some static API, as the result, there is no convention about what `by` is. If it's a column name, should it be just a name or the query compiler of this column? For now, it could be anything. There are certain cases (reflected in [these](https://github.com/modin-project/modin/blob/3f0e39e382fac347240b3c081ade066dc2d6dc38/modin/pandas/test/test_groupby.py#L1410) tests), when backends obtain `by` as a list of column names, and many other cases, when it obtains `by` as a query compiler. Currently, the problem with missed columns at the backend is handled by [this clumsy construction](https://github.com/modin-project/modin/blob/3f0e39e382fac347240b3c081ade066dc2d6dc38/modin/pandas/groupby.py#L482-L519) that I would like to get rid of.

My first thought was, that cases with column names at the backend is a bug, and it just needed to be fixed, however at #2744, I saw that `ModinDataFrame.groupby` obtains column names as `by` [[link]](https://github.com/RehanSD/modin/blob/d31e9640c205489716f9bcb61d61ee82e79bfb44/modin/engines/modin_dataframe.py#L233).

So firstly we need to decide what `by` is at the backend, and based on this decide where the logic about grabbing the first column at `groupby.size` should be located.